plugins/provisioners/puppet/provisioner/puppet.rb

```diff
@@ -28,18 +28,12 @@ def configure(root_config)
             @module_paths << [path, File.join(config.temp_dir, "modules-#{key}")]
           end
 
-          folder_opts = {}
-          folder_opts[:type] = @config.synced_folder_type if @config.synced_folder_type
-          folder_opts[:owner] = "root" if !@config.synced_folder_type
-          folder_opts[:args] = @config.synced_folder_args if @config.synced_folder_args
-          folder_opts[:nfs__quiet] = true
-
           if @config.environment_path.is_a?(Array)
             # Share the environments directory with the guest
             if @config.environment_path[0].to_sym == :host
               root_config.vm.synced_folder(
                 File.expand_path(@config.environment_path[1], root_path),
-                environments_guest_path, folder_opts)
+                environments_guest_path, @config.synced_folder_opts)
```

Comment:
These calls i really would like to test with some sort of integration test, at least against a mocked ```root_config.vm.synced_folder``` method, but did not find any good copy-paste template for it so to understand what i have to do here. Maybe you know of some place i can peak a solution to test this?

Reply:
You should be able to mock this up with Rspec. Unfortunately the puppet provisioner is very sparse on tests :( You'll want to create a new describe `context` for this classes rspec tests around the `#configure` method. Ideally here I would want to mock up that a given _double_ called `vm` received the method `:synced_folder` with the specific config options. That will be good enough locally for unit tests.

Something like

```ruby
describe "#configure" do
  let(:vm) { double("vm") }
  let(:root_config) { double("root_config" } # This should actually be real when writing this test, not just pseudo code
...
  it "configures the provisioner" do
    # other stuff probably goes here!!
    expect(vm).to receive(:synced_folder).with("/path/to/the/manifest", {type=>"nfs", other=>"keys"})

    subject.configure(root_config)
  end
end
```

This is super basic and missing a lot of what would probably be required :sweat_smile: But that is basically how I would test that the _right_ thing is being passed to the `#synced_folder` function to be stored as a config.

Reply:
You will also probably need to ensure that the double `vm` is the right class and not just a double? Since it's apart of `root_config`, you'll likely need to do a bit of mocking to ensure that it is a `VagrantPlugins::Kernel_V2::VMConfig` type. But anyway, hopefully that gives you a good idea :eyes:

Reply:
Yes, exactly :) Thanks very much, because it would have been tedious to search this together without knowing what to do in a new language :) Very much appreciated :+1: